tt_metal/impl/program/program.hpp

```diff
@@ -133,6 +133,7 @@ class Program {
     // The buffer that holds the kernel/binaries/etc for this program
     std::vector<std::unique_ptr<Buffer>> kg_buffers;
     std::unique_ptr<Buffer> buffer;
+    bool runtime_args_dirty;
```

Comment:
what sets this when runtime args are updated by user app?

Reply:
I think the idea is that when runtime args are updated, we can reuse the transfer_infos that are populated when the program was created with `runtime_args_dirty=true`.

Reply:
yes, really there should be 2 levels of dirty. right now we'll always re-populate the data but we don't rebuild the structure.